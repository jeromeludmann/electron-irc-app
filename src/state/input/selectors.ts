import { createSelector } from 'reselect'
import { getBufferFactory } from '@app/state/buffer/selectors'
import { Route } from '@app/utils/Route'

export function getInput(route?: Route) {
  return createSelector(getBufferFactory(route), channel => channel.input)
}

export function getInputValue(route?: Route) {
  return createSelector(getInput(route), input => input.value)
}
