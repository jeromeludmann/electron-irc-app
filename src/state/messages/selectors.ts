import { createSelector } from 'reselect'
import { getBufferFactory } from '@app/state/buffer/selectors'
import { Route } from '@app/utils/Route'

export function getMessages(route?: Route) {
  return createSelector(getBufferFactory(route), buffer => buffer.messages)
}
