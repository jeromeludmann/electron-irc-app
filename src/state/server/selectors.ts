import { createSelector } from 'reselect'
import { getServers } from '@app/state/root/selectors'
import { getRoute } from '@app/state/route/selectors'

export function getServerFactory(serverKey?: string) {
  return createSelector(
    getServers,
    getRoute,
    (servers, route) => servers[serverKey || route.serverKey],
  )
}

export function getBuffers(serverKey?: string) {
  return createSelector(getServerFactory(serverKey), server => server.buffers)
}
