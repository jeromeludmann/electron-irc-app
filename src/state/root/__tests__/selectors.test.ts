import { rootInitialState } from '@app/state/root/reducer'
import { getServers, getServerKeys } from '@app/state/root/selectors'

describe('getServers()', () => {
  it('should get servers', () => {
    expect(getServers(rootInitialState)).toMatchSnapshot()
  })
})

describe('getServerKeys()', () => {
  it('should get server keys', () => {
    expect(getServerKeys(rootInitialState)).toMatchSnapshot()
  })
})
