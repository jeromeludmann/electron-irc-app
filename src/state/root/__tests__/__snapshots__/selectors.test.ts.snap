// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getServerKeys() should get server keys 1`] = `
Array [
  "serverKey",
]
`;

exports[`getServers() should get servers 1`] = `
Object {
  "serverKey": Object {
    "buffers": Object {
      "@raw": Object {
        "activity": false,
        "input": Object {
          "dirtyValue": "",
          "history": Object {
            "index": 0,
            "values": Array [],
          },
          "value": "",
        },
        "messages": Array [],
      },
      "@status": Object {
        "activity": false,
        "input": Object {
          "dirtyValue": "",
          "history": Object {
            "index": 0,
            "values": Array [],
          },
          "value": "",
        },
        "messages": Array [],
      },
    },
    "lag": 0,
    "modes": Object {
      "available": Object {
        "channel": Array [],
        "user": Array [],
      },
      "user": Array [],
    },
    "name": "<unknown>",
    "user": Object {
      "nick": "default_nick",
      "real": "default_name",
      "user": "default_user",
    },
  },
}
`;
